server/index.js

```diff
@@ -147,6 +147,29 @@ app.post('/api/cart', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/orders', (req, res, next) => {
+  if (!req.session.cartId) {
+    return next(new ClientError('No order is currently available', 400));
+  }
+  const hasValidOrderParams = req.body.name && req.body.creditCard && req.body.shippingAddress;
+  if (!hasValidOrderParams) {
+    return next(new ClientError('Order details are not valid. Please provide a name, a credit card number, and a shipping address for your order.', 400));
+  }
+
+  const query = `
+     INSERT INTO "orders" ("cartId", "name", "creditCard", "shippingAddress")
+          VALUES ($1, $2, $3, $4)
+          RETURNING "orderId", "cartId", "name", "creditCard", "shippingAddress";
```

Comment:
We want the timestamp instead of the `cartId`

Reply:
Resolved in 10e9dcd.